### PR TITLE
F rename cols

### LINKED ISSUE #41: Implement cdm_rename()
Required for #40. Same syntax as `cdm_filter()` .

`cdm_select()` is similar but has the additional complexity that removed columns need to be taken care of. Later.

### PR DESCRIPTION
**Pull request by TSchiefer**
- function renames columns of tables, taking care of keys

closes #41

### CODE REVIEW

File: R/data-model-helpers.R [hunk 1]
```diff
@@ -245,3 +245,50 @@ data_model_db_types_to_R_types <- function(data_model) {
   data_model$columns$type <- new_type
   data_model
 }
+
+upd_pks_after_rename <- function(
+  pks,
```
Comment (krlmlr) [inline R/data-model-helpers.R]: This needs to come right after `function(` .

File: R/data-model-helpers.R [hunk 2]
```diff
@@ -245,3 +245,50 @@ data_model_db_types_to_R_types <- function(data_model) {
...
+  table_name,
+  list_of_renames) {
+
+  pk_of_table <- filter(pks, table == !!table_name) %>% pull(column)
```
Comment (krlmlr) [inline R/data-model-helpers.R]: We could also express with `left_join()` and `coalesce()` : 

```r
rename_logic <- ...
new_pks <-
  pks %>%
  left_join(rename_logic) %>%
  mutate(table = coalesce(new_table, table))
```

No need to spend time here, though.

File: R/data-model-helpers.R [hunk 3]
```diff
@@ -245,3 +245,50 @@ data_model_db_types_to_R_types <- function(data_model) {
...
+  if (any(list_of_renames == pk_of_table)) {
+    new_pk <- names(list_of_renames[list_of_renames == pk_of_table])
+    pks$column[pks$table == table_name] <- new_pk
+  }
+  pks
+}
+
+upd_fks_after_rename <- function(
```
Comment (krlmlr) [inline R/data-model-helpers.R]: This is where I think a longer-form data structure with three columns (table, column, role) would fare much better: we'd only need one update routine. TBD.

File: R/dplyr-dm.R [hunk 4]
```diff
@@ -0,0 +1,65 @@
+#' Rename one or more columns of a [`dm`] table
+#'
+#' Rename columns of your [`dm`] with a similar syntax to `dplyr::rename()`.
+#'
+#' @inheritParams cdm_filter
+#' @param ... One or more unquoted expressions separated by commas. You can treat
```
Comment (krlmlr) [inline R/dplyr-dm.R]: Instead of copy-pasting, would it work to use `@inheritParams dplyr::select` ?

File: R/dplyr-dm.R [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+  fks_upd <-
+    upd_fks_after_rename(
+      cdm_get_data_model_fks(dm),
+      table_name,
+      list_of_renames
+    )
+
+  new_dm2(
+    src = cdm_get_src(dm),
```
Comment (krlmlr) [inline R/dplyr-dm.R]: Covered by `base_dm` ?

File: R/dplyr-dm.R (hunk 4 above)
Comment (TSchiefer) [reply R/dplyr-dm.R]: I slightly changed the text here and there, 2 reasons:
1. We don't have `cdm_select()` yet
2. Some things would not be quite right

don't have a strong opinion though

File: R/data-model-helpers.R (hunk 1 above)
Comment (TSchiefer) [reply R/data-model-helpers.R]: This as in `pks,`?

File: R/data-model-helpers.R [hunk 6]
```diff
@@ -245,3 +245,50 @@ data_model_db_types_to_R_types <- function(data_model) {
...
+  fks,
+  table_name,
+  list_of_renames) {
+
+  fks_from_table <- filter(fks, table == !!table_name) %>% pull(column)
+  if (!is_empty(fks_from_table) &&
```
Comment (krlmlr) [inline R/data-model-helpers.R]: Can you extract this and the block below into a function?

**Pull request merged**